File: CppPractice/CppLeetcode/Leetcode46.cpp
#include<vector>
using namespace std;

class Solution {
private:
    void helper(vector<int> &nums,vector<int> &temp,vector<vector<int>> &result)
    {
        if (temp.size() == nums.size())
        {
            result.push_back(temp);
        }
        for (int i = 0; i < nums.size(); i++)
        {
            if (find(temp.begin(),temp.end(),nums[i])==temp.end())
            {
                temp.push_back(nums[i]);
                helper(nums,temp,result);
                temp.pop_back();
            }
        }
    }
public:
    vector<vector<int>> permute(vector<int>& nums) 
    {
        vector<vector<int>> result;
        vector<int> temp;
        
        helper(nums,temp,result);

        return result;
    }
};

File: JavaScriptPractice/Leetcode46.js
let result;
let temp;
let numsLen;

function helper(nums)
{
    if (temp.length==numsLen) 
    {
        result.push([...temp]);
    }
    for (let i = 0; i < nums.length; i++) 
    {
        if (!temp.includes(nums[i])) 
        {
            temp.push(nums[i]);
            helper(nums);
            temp.pop();
        }
    }
}
var permute = function(nums) 
{
    result=[];
    temp = [];
    numsLen = nums.length;
    helper(nums)

    return result;
    
};